### PR TITLE
[Merged by Bors] - Docs: Split Usage Guide into individual files

### PR DESCRIPTION
**Pull request by fhennig**
# Description

This is something we want to do in all operators eventually.

I've also removed obsolete setup instructions, as they are already covered in the Getting Started Guide

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/usage_guide/pages/security.adoc [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Authorization is not fully implemented by the Stackable Operator for Apache Nifi.
+
+=== Single user
+
+With this authorization method, a single user has administrator capabilities.
+
+[#authorization-ldap]
+=== LDAP
+
+The operator uses the https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#fileusergroupprovider[`FileUserGroupProvider`] and https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#fileaccesspolicyprovider[FileAccessPolicyProvider] to bind the LDAP user to the Nifi administrator group. This user is then able to create and modify groups and polices in the web interface. These changes local to the `Pod` running Nifi and are *not* persistent.
```
Comment (maltesander) [inline docs/modules/usage_guide/pages/security.adoc:78]:
```suggestion
Authorization is not fully implemented by the Stackable Operator for Apache NiFi.

=== Single user

With this authorization method, a single user has administrator capabilities.

[#authorization-ldap]
=== LDAP

The operator uses the https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#fileusergroupprovider[`FileUserGroupProvider`] and https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#fileaccesspolicyprovider[FileAccessPolicyProvider] to bind the LDAP user to the NiFi administrator group. This user is then able to create and modify groups and polices in the web interface. These changes local to the `Pod` running NiFi and are *not* persistent.
```

File: docs/modules/usage_guide/pages/security.adoc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  [...]
+  config:
+    authentication:
+      method:
+        singleUser:
+          adminCredentialsSecret: nifi-admin-credentials-simple # <1>
+          autoGenerate: true
+----
+
+<1> Administrator credentials for logging into the Nifi web interface. This is the name of a `Secret` resource with two fields: `username` and `password`. This `Secret` must exist but it's entries can be populated by the operator when `autoGenerate` is `true`.
```
Comment (maltesander) [inline docs/modules/usage_guide/pages/security.adoc:36]:
```suggestion
<1> Administrator credentials for logging into the NiFi web interface. This is the name of a `Secret` resource with two fields: `username` and `password`. This `Secret` must exist but it's entries can be populated by the operator when `autoGenerate` is `true`.
```

File: docs/modules/usage_guide/pages/resource-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
+
+= Resource configuration
+
+== Volume storage
+
+By default, a Nifi cluster will create five different persistent volume claims for flow files, provenance, database, content and state folders. These PVCs will request `2Gi`. It is recommended that you configure these volume requests according to your needs.
```
Comment (maltesander) [inline docs/modules/usage_guide/pages/resource-configuration.adoc:6]:
```suggestion
By default, a NiFi cluster will create five different persistent volume claims for flow files, provenance, database, content and state folders. These PVCs will request `2Gi`. It is recommended that you configure these volume requests according to your needs.
```

File: docs/modules/usage_guide/pages/resource-configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+              capacity: 12Gi
+----
+
+In the above example, all nodes in the default group will request `12Gi` of storage the various folders.
+
+== Resource Requests
+
+include::home:concepts:stackable_resource_requests.adoc[]
+
+If no resource requests are configured explicitly, the Nifi operator uses the following defaults:
```
Comment (maltesander) [inline docs/modules/usage_guide/pages/resource-configuration.adoc:36]:
```suggestion
If no resource requests are configured explicitly, the NiFi operator uses the following defaults:
```

File: docs/modules/usage_guide/pages/configuration-environment-overrides.adoc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+IMPORTANT: Do not override port numbers.
+This will lead to cluster malfunction.
+
+== Configuration Overrides
+
+Apache NiFi runtime configuration is stored in the files bootstrap.conf and nifi.properties.
+The `configOverrides` block enables you to customize parameters in these files.
+The complete list of the configuration options can be found in the  https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#system_properties[Apache NiFi documentation].
+
+Overrides are key, value pairs defined under a Nifi configuration file such as `bootstrap.conf` or `nifi.properties`. They must match the names values as expected by NiFi. In the example below, a property `nifi.flow.configuration.archive.enabled` is being explicitly set to 'false', overriding the default value.
```
Comment (maltesander) [inline docs/modules/usage_guide/pages/configuration-environment-overrides.adoc:14]:
```suggestion
Overrides are key, value pairs defined under a NiFi configuration file such as `bootstrap.conf` or `nifi.properties`. They must match the names values as expected by NiFi. In the example below, a property `nifi.flow.configuration.archive.enabled` is being explicitly set to 'false', overriding the default value.
```

File: docs/modules/usage_guide/pages/security.adoc [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+<1> The reference to an AuthenticationClass called `ldap`
+
+You can follow the xref:nightly@home:tutorials:authentication_with_openldap.adoc[] tutorial to learn how to set up an AuthenticationClass for an LDAP server, as well as consulting the xref:nightly@home:reference:authenticationclass.adoc[] reference.
+
+== Authorization
+
+NiFi supports multiple authorization methods documented https://nifi.apache.org/docs/nifi-docs/html/administration-guide.html#multi-tenant-authorization[here].
+The available authorization methods depend on the chosen authentication method.
+
+Authorization is not fully implemented by the Stackable Operator for Apache Nifi.
```
Comment (maltesander) [inline docs/modules/usage_guide/pages/security.adoc:69]:
```suggestion
Authorization is not fully implemented by the Stackable Operator for Apache NiFi.
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Just for consistency. LGTM otherwise!

**Comment by fhennig** [comment]
bors merge

**Pull request closed without merging**